File: JPA/studyolle/src/main/java/me/kzv/olle/account/AccountRepository.java
package me.kzv.olle.account;

import me.kzv.olle.domain.Account;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.transaction.annotation.Transactional;


@Transactional(readOnly = true) // 성능의 이점?을 갖기 위해
public interface AccountRepository extends JpaRepository<Account, Long> {
    boolean existsByEmail(String email);

    boolean existsByNickname(String nickname);
}

File: toy/studyolle/src/main/kotlin/me/kzv/study/account/AccountRepository.kt
package me.kzv.study.account

import me.kzv.study.domain.Account
import org.springframework.data.jpa.repository.JpaRepository

interface AccountRepository : JpaRepository<Account, Long> {
    fun existsByEmail(email: String): Boolean
    fun existsByNickname(email: String): Boolean
}